src/nvim/eval.c

```diff
@@ -6007,6 +6012,26 @@ static void rettv_dict_alloc(typval_T *rettv)
   ++d->dv_refcount;
 }
 
+/// Clear the all the keys of a Dictionary.
+/// "d" remains a valid empty Dictionary.
+///
+/// @param d The Dictionary to clear
+void dict_clear(dict_T *d) {
```

Comment:
Also missing FUNC_ATTR_NONNULL_ALL (or, better, `NULL` handling because `NULL` is a valid dictionary).